### PR TITLE
[SigV4] Add assume role support when using EC2 IAM auth

### PR DESCRIPTION
**Pull request by wbrowne**

### CODE REVIEW

File: pkg/sigv4/sigv4.go [hunk 1]
```diff
@@ -151,6 +151,18 @@ func (m *middleware) signer() (*v4.Signer, error) {
 		}
 		c = credentials.NewCredentials(&ec2rolecreds.EC2RoleProvider{Client: ec2metadata.New(s), ExpiryWindow: stscreds.DefaultDuration})
 
+		if m.config.AssumeRoleARN != "" {
+			s, err = session.NewSession(&aws.Config{
+				CredentialsChainVerboseErrors: aws.Bool(true),
```
Comment (sunker) [inline pkg/sigv4/sigv4.go:156]: Do we want to keep this?

**Review by sunker — APPROVED** [review]
LGTM! 

I guess we can remove line 193-202 now?

File: pkg/sigv4/sigv4.go (hunk 1 above)
Comment (wbrowne) [reply pkg/sigv4/sigv4.go:156]: Yeah I believe so. This will help give more useful error messages. It's something we already do for sessions.

**Comment by wbrowne** [comment]
> LGTM!
> 
> I guess we can remove line 193-202 now?

Yeah these lines are still relevant for the `AuthTypeKeys` and `AuthTypeSharedCreds` cases where we only set the credentials

**Comment by sunker** [comment]
> > LGTM!
> > I guess we can remove line 193-202 now?
> 
> Yeah these lines are still relevant for the `AuthTypeKeys` and `AuthTypeSharedCreds` cases where we only set the credentials

Okay cool!

**Comment by wbrowne** [comment]
Tested and looks good!

**Pull request merged**